homeworks/solutions/torantie/3-smooth/README.md

```diff
@@ -0,0 +1,97 @@
+# Homework 3: Laplacian Smoothing
+
+In this homework, your task is to implement the Laplacian smoothing algorithm
+for the **uniform** and the **cotan** Laplace-Beltrami operator.
+
+## Skeleton
+
+The skeleton for this homework is structured as follows:
+
+```
+3-smooth
+├── package.json
+├── package-lock.json
+├── README.md             <--- You work on this file
+├── src
+│   ├── assets
+│   │   └── bunny_tri.obj
+│   ├── halfedge.js       <--- You work on this file
+│   ├── main.js
+│   ├── renderer.js
+│   └── vec.js
+└── webpack.config.js
+```
+
+You coding tasks are located in the `halfedge.js`.
+Please look for `TODO:` in these two files and complete them to archive
+the reference results listed below.
+
+> It might be useful to read the code in `main.js`.
+
+You should also document your implementation process. Specifically, you
+should answer the following questions:
+
+1. **Implementation complexity**: Which code snippet (report in line numbers) in the `halfedge.js` is the most time consuming for you to implement? Explain your coding experience and encountered challenges briefly.
+
+```
+TODO: your answer goes here
+```
+
+line 562-631 the smooth function: Translating the linear equation into code was very time consuming because i didn't understand or find what some things of the equations ment in the slides. Also debugging the Matrices was extremely hard because you couldn't print sparse matrices.
```

Comment:
Good point. In fact, you could try to convert a matrix to DenseMatrix, then use get(i, j) for checking the matrix elements.